### PR TITLE
Fix: use configurationChanges field on deploy

### LINKED ISSUE #49: Use create environment API instead of create + set config + deploy
Create Environment API already supports `deployRequest` and `configurationChanges` in body so we should use this single API on deployment instead of doing 3 different API calls.

### PR DESCRIPTION
**Pull request by yaronya**
Fixes #49

### Issue & Steps to Reproduce
Align the CLI with how the UI works on initial deploy + redeploy.

### Solution
1. Use `deployRequest` field on initial deloy.
2. Use `configurationChanges` to set proper `DEPLOYMENT` configuration scope on both initial deploy and redeploy.

### CODE REVIEW

File: node/src/commands/deploy.js [hunk 1]
```diff
@@ -2,41 +2,41 @@ const DeployUtils = require('../lib/deploy-utils');
...
-      );
-      await deployUtils.setConfiguration(environment, blueprintId, config.name, config.value, config.sensitive);
-    }
-  }
-};
+const { BLUEPRINT_ID, REVISION, ENVIRONMENT_NAME, PROJECT_ID } = options;
 
 const assertBlueprintExistsOnInitialDeployment = options => {
   if (!options[BLUEPRINT_ID]) throw new Error('Missing blueprint ID on initial deployment');
+  if (!options[REVISION]) throw new Error('Missing revision on initial deployment');
```
Comment (avnerenv0) [inline node/src/commands/deploy.js:9]: Why?
We don't have to have a revision, every git repo has a default branch.

File: node/src/commands/deploy.js [hunk 2]
```diff
@@ -2,41 +2,41 @@ const DeployUtils = require('../lib/deploy-utils');
...
+
+  let deployment;
+  let environment = await deployUtils.getEnvironment(options[ENVIRONMENT_NAME], options[PROJECT_ID]);
 
   if (!environment) {
+    logger.info('Initial deployment detected');
     assertBlueprintExistsOnInitialDeployment(options);
-    environment = await deployUtils.createEnvironment(environmentName, organizationId, projectId);
+    environment = await deployUtils.createAndDeployEnvironment(options, configurationChanges);
+    deployment = await deployUtils.getDeployment(environment.latestDeploymentLogId);
```
Comment (avnerenv0) [inline node/src/commands/deploy.js:35]: this is not needed, we are only using the `id` anyway

File: node/src/lib/genetal-utils.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
-const convertRequiresApprovalToBoolean = str => {
+const convertStringToBoolean = str => {
   if (str === 'true') return true;
   if (str === 'false') return false;
 };
 
+const removeEmptyValuesFromObj = payload =>
+  JSON.parse(
+    JSON.stringify(payload, (_, value) => {
+      if (value === undefined) return undefined;
```
Comment (avnerenv0) [inline node/src/lib/genetal-utils.js:9]: this function always returns `value` :)
You can `JSON.parse(JSON.stringify(payload));`

**Review by avnerenv0 — APPROVED** [review]
@yaronya left you some comments,
except the `revision` thing, nothing is important 

Makes me understand that any configuration changes given to the deployment API should be scoped as 'DEPLOYMENT' by the backend anyway, the clients don't need to know about this.

File: node/src/commands/deploy.js (hunk 1 above)
Comment (yaronya) [reply node/src/commands/deploy.js]: I looked only on `create` where it gets it from request but then on `deploy` it's not required 🤦 
64f9e72

File: node/src/commands/deploy.js (hunk 2 above)
Comment (yaronya) [reply node/src/commands/deploy.js]: d7cec36

**Comment by yaronya** [comment]
> Makes me understand that any configuration changes given to the deployment API should be scoped as 'DEPLOYMENT' by the backend anyway, the clients don't need to know about this.

@avnerenv0 TBH It's already scoped by the backend but yet I added it to the request. I sense that this might be confusing. I've removed it from request: 0e5aa01

**Review by roni-frantchi — APPROVED** [review]

**Pull request merged**